cdc/entry/mounter.go

```diff
@@ -386,52 +391,80 @@ func (m *mounterImpl) mountRowKVEntry(tableInfo *TableInfo, row *rowKVEntry) (*m
 			whereHandle := true
 			col.WhereHandle = &whereHandle
 		}
-		values[colName] = col
+		cols[colName] = col
+	}
+	for _, col := range tableInfo.Columns {
+		_, ok := cols[col.Name.O]
+		if !ok && tableInfo.IsColWritable(col) {
+			column := &model.Column{
+				Type:  col.Tp,
+				Value: getDefaultOrZeroValue(col),
+				Flag:  transColumnFlag(col),
+			}
+			if tableInfo.IsColumnUnique(col.ID) {
+				whereHandle := true
+				column.WhereHandle = &whereHandle
+			}
+			cols[col.Name.O] = column
+		}
+	}
+	return cols, nil
+}
+
+func (m *mounterImpl) mountRowKVEntry(tableInfo *TableInfo, row *rowKVEntry) (*model.RowChangedEvent, error) {
+	if !m.enableOldValue && row.Delete && !tableInfo.PKIsHandle {
+		return nil, nil
+	}
+
+	var err error
+	// Decode previous columns.
+	var preCols map[string]*model.Column
+	if len(row.PreRow) != 0 {
+		preCols, err = datum2Column(tableInfo, row.PreRow)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	// Try to decode columns, if row represents a delete event, columns should be == previous columns
+	var cols map[string]*model.Column
+	if row.Delete && m.enableOldValue {
+		// if row is Delete and old value was enabled.
+		cols = preCols
+		preCols = nil
+	} else {
+		cols, err = datum2Column(tableInfo, row.Row)
```

Comment:
if the `row.Delete` is true and the `m.enableOldValue` is false; the delete event should only include the handle column